### ISSUE TITLE
Feature Request: Inline Preview Rendering/Editing Like Typora

### ISSUE THREAD

**Issue by alexheyd** [shd101wyy/markdown-preview-enhanced#425]
First of all, thanks for such a great package. It has become my preferred way of writing Markdown and creating slides.

I recently came across Typora (https://typora.io) and was really interested in their approach. They forego the preview panel altogether and allow you to edit the markdown in the rendered view.

Would this be something you'd be interested in adding as a setting that can be toggled?

**Comment by shd101wyy** [comment]
Hi @alexheyd ,

Haha I have also thought about it and tried to make a clone of Typora in [branch project-mars](https://github.com/shd101wyy/markdown-preview-enhanced/tree/project-mars). However it is extremely hard and the WYSIWYG way of editing markdown is super difficult to implement. I think the author of Typora really puts tons of efforts into that project.  

Anyways, your suggestion will always be in my plans. I will notice you if one day I successfully make it.  

Thanks

**Comment by alexheyd** [comment]
Thanks for the quick response. Yeah I suspected it'd be a rather large addition. FWIW, in case you haven't come across this post, I found this related to Typora's markdown engine: https://github.com/typora/typora-issues/issues/315

Perhaps looking at draft.js or codemirror will help inform the implementation.

I'm not super familiar with CoffeeScript but I'm a JS dev, so I can poke around `project-mars` and see if I can help in anyway.

**Comment by xnervwang** [comment]
Please consider it seriously. I don't think the style of Typora is suitable for everyone. I once used Typora and I'd like to use it as a preview tool for Markdown, not a edit tool. So is the reason why I prefer to Atom markdown.
I guess most people use markdown instead of Some rich text editor like Word, because they want to focus on the written content, and they don't like to consider some problems like "why is there space here?" "why inter-line space in these lines is larger than those lines?"...So they use markdown since they know what they see will be what they write.
Typora seems good, but not everyone likes its style, like me. I think it's just a lite version for Word.

**Comment by shd101wyy** [comment]
Don't worry, I won't remove the source/preview way of editing markdown.

**Comment by Piping** [comment]
@xnervwang you know typora has a shortcut `cmd+\` to view plain text right?

**Comment by xnervwang** [comment]
@Piping

Yeah I know. I even used a mouse hotkey to switch the view when I didn't like to edit markdown just like editing web page in FrontPage. For myself, the best benifit of Typora is it could use the whole screen as preview and I can switch to plain mode to modify some tiny text places. But as a tool for sorting notes, I prefer to MPE with treeview.

I suggest author could make the preview page editable instead of readonly to fit the requirement.

**Comment by 625781186** [comment]
@shd101wyy 
Hi , I found a repository  [usecanvas/canvas-editor](https://github.com/usecanvas/canvas-editor) .
This is  [GIF](https://github.com/625781186/todo_ninjia3_repository/blob/master/static/ScreenShot.gif) animation running the repository above.
This is a [vedio](https://www.youtube.com/watch?v=b0fqskVXUMY).

Hope to inspire you.

我去 , 打了这么多字发现你是国人!

**Comment by mitinarseny** [comment]
@alexheyd Any updates on this?

For me, it is not necessary to implement WYSIWYG fully, but inline KaTeX rendering and would be awesome!

**Comment by shd101wyy** [comment]
Yes I have been playing with a very cool WYSIWYG editor called [HyperMD](https://github.com/laobubu/HyperMD) recently. It seems to be a good choice to start with instead of me implementing everything from scratch.

**Comment by mitinarseny** [comment]
Hmm, seems nice, I'l take a look

But the key feature for me in VS Code is that I can use pandoc. Unfortunately, as I can see from HyperMD documentation, there is nothing there about pandoc.
